sql-plugin/src/main/spark400/scala/com/nvidia/spark/rapids/shims/GpuBatchScanExec.scala

```diff
@@ -47,7 +47,7 @@ case class GpuBatchScanExec(
   @transient override lazy val batch: Batch = if (scan == null) null else scan.toBatch
   // TODO: unify the equal/hashCode implementation for all data source v2 query plans.
   override def equals(other: Any): Boolean = other match {
-    case other: GpuBatchScanExec =>
+    case other: BatchScanExec =>
```

Comment:
Why is this being changed? This is a GpuBatchScanExec. We don't want to be equal to non-GPU versions do we?

Reply:
Right. While debugging I wasn't sure what was causing a failure and looking at 330 [shim](https://github.com/NVIDIA/spark-rapids/blob/7de3fc4a9b23b819be632c6cfea85a706cdafef3/sql-plugin/src/main/spark330/scala/com/nvidia/spark/rapids/shims/GpuBatchScanExec.scala#L49) I changed this and didn't change it back before submitting this PR. 

I am adding that change to this PR as well.